analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SolutionBuilder.cs

```diff
@@ -78,10 +79,17 @@ internal struct SolutionBuilder
                 throw new ArgumentException("Please use a collection of paths with the same extension", nameof(paths));
             }
 
-            var project = Create()
-                .AddProject(AnalyzerLanguage.FromPath(paths.First()), outputKind: outputKind)
-                .AddDocuments(paths)
-                .AddReferences(additionalReferences);
+            var project = duplicateDocumentsNeeded switch
+            {
```

Comment:
2 out of 3 actions are duplicated here and we have builder => let's do the two actions in series and then decide with `if` what do do differently, like the unmodified statements below

Reply:
Logic was moved. No longer needed.